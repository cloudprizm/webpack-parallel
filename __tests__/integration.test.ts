import { runWebpackConfigs } from "../dist"
import { join, resolve } from "path"
import { bootstrap } from "@hungry/babel-preset-cli"

const a = require('@hungry/babel-preset-cli')
console.log('@@@@', Object.keys(a))
const workerFile = join("../dist", "./worker-babel-wrapper.js")
const exit = jest.spyOn(process, "exit").mockImplementation(n => n)

test("full run test", done => {
  const unhook = bootstrap()
  const config = resolve(__dirname, "./fixtures/configs.ts")

  const compile = runWebpackConfigs({
    silent: true,
    watch: false,
    cwd: __dirname,
    config,
    workerFile,
    fullReport: false
  })

  compile.then((stats) => {
    const ignorePid = stats.map((d, idx) => ({ ...d, pid: idx }))

    expect(ignorePid).toMatchSnapshot()
    unhook()

    setImmediate(() => {
      expect(exit).toHaveBeenCalledWith(0)
      done()
    })
  })
})