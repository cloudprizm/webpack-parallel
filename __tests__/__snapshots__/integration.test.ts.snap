// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full run test 1`] = `
Array [
  Object {
    "action": "end",
    "errors": Array [],
    "filteredModules": 1,
    "id": 0,
    "idx": 0,
    "modules": Array [],
    "name": "C0",
    "pid": 0,
    "warnings": Array [],
  },
  Object {
    "action": "end",
    "errors": Array [],
    "filteredModules": 1,
    "id": 1,
    "idx": 1,
    "modules": Array [],
    "name": "C1",
    "pid": 1,
    "warnings": Array [],
  },
]
`;
